clients/shared/config/apiConfig.ts

```diff
@@ -0,0 +1,24 @@
+import { fetchAuthSession } from '@aws-amplify/auth'
+import axios from 'axios'
+
+const api = axios.create({
+  baseURL: import.meta.env.VITE_API_URL,
+  headers: {
+    'Content-Type': 'application/json'
+  }
+})
+
+api.interceptors.request.use(
+  async(config) => {
+    const idToken = (
+      await fetchAuthSession()
+    ).tokens?.idToken?.toString()
+    if (idToken !== undefined && idToken !== null) {
+      config.headers.Authorization = `Bearer ${idToken}`
+    }
+    return config
+  },
+  async(error) => await Promise.reject(error)
```

Comment:
no need to await

Reply:
updated in a3d06124eeaeb1aff9ed3fd02c4eadf7f7467947